core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -93,41 +106,53 @@ object Broker {
    *               id (int), number of endpoints (int), serialized endpoints
    * @return broker object
    */
-  def readFrom(buffer: ByteBuffer): Broker = {
+  def readFrom(buffer: ByteBuffer, expectRack: Boolean = true): Broker = {
```

Comment:
It seems that readFrom and Broker.writeTo are only used in tests now since the serialization of UpdateMetadataRequest is based on the one in o.a.k. Instead of maintaining the logic here, could we just remove readFrom and Broker.writeTo and the corresponding test code?